tools/genesis-gen/src/gen.rs

```diff
@@ -0,0 +1,586 @@
+        keys: Vec<KeyPair>,
+    }
+
+    impl From<Account> for NewAccount {
+        fn from(Account { id, keys }: Account) -> Self {
+            OriginAccount::new(id, keys.iter().map(|x| x.public_key().clone()))
+        }
+    }
+
+    impl Account {
+        fn new(rnd: &mut RandHelp, domain_id: &DomainId) -> Result<Self> {
+            let kp: KeyPair = {
+                let seed_len: usize = rnd.rng.gen_range(5..30);
+                let mut seed: Vec<u8> = Vec::with_capacity(seed_len);
+                for _ in 0..seed_len {
+                    seed.push(rnd.rng.gen());
+                }
+
+                let config = KeyGenConfiguration::default().use_seed(seed);
+                KeyPair::generate_with_configuration(config)?
+            };
+            let keys = vec![kp];
+
+            let id = Self::gen_account_id(rnd, domain_id)?;
+
+            Ok(Self { id, keys })
+        }
+
+        fn gen_account_id(rnd: &mut RandHelp, domain_id: &DomainId) -> Result<AccountId> {
+            try_gen_rand_non_repetitive_value(&mut rnd.used_names.accounts, || {
+                let name = rnd.petnames.generate(&mut rnd.rng, 3, "_");
+                let name = construct_iroha_name(&name)?;
+                let id = AccountId::new(name, domain_id.clone());
+                let id_str = id.to_string();
+                Ok((id_str, id))
+            })
+            .wrap_err("Failed to generate account id")
+        }
+    }
+
+    struct AssetDefinition {
+        id: AssetDefinitionId,
+        value_type: AssetValueType,
+        mintable: Mintable,
+    }
+
+    impl From<AssetDefinition> for OriginAssetDefinition {
+        fn from(
+            AssetDefinition {
+                id,
+                value_type,
+                mintable,
+            }: AssetDefinition,
+        ) -> Self {
+            use AssetValueType::{BigQuantity, Fixed, Quantity, Store};
+            use Mintable::{Infinitely, Not, Once};
+
+            let mut definition = match value_type {
+                Quantity => OriginAssetDefinition::quantity(id),
+                BigQuantity => OriginAssetDefinition::big_quantity(id),
+                Fixed => OriginAssetDefinition::fixed(id),
+                Store => OriginAssetDefinition::store(id),
+            };
+
+            let definition = match mintable {
+                Not => {
+                    definition.mintable(false);
+                    definition
+                }
+                Once => definition.mintable_once(),
+                Infinitely => {
+                    definition.mintable(true);
+                    definition
+                }
+            };
+
+            definition.build()
+        }
+    }
+
+    impl AssetDefinition {
+        fn new(rnd: &mut RandHelp, domain_id: &DomainId) -> Result<Self> {
+            let mintable: Mintable = match rnd.rng.gen_range(0..10u32) {
+                5..=9 => Mintable::Infinitely,
+                2..=4 => Mintable::Not,
```

Comment:
Not an accurate representation. Assets have to be minted once in order to be non-zero. If an asset has zero presence it gets removed from active definitions. Please just merge with `Mintable::Once`.

Reply:
Fixed